File: core/src/com/xenoage/zong/core/music/direction/Wedge.java
package com.xenoage.zong.core.music.direction;

import lombok.Data;
import lombok.EqualsAndHashCode;


/**
 * Class for crescendos and diminuendos.
 * 
 * To create a wedge, create an instance of this class,
 * retrieve its end marker by calling {@link #getWedgeEnd()} and
 * place it anywhere after the instance of this class. The wedge ends
 * at the element at which the end marker is placed.
 * 
 * @author Andreas Wenger
 */
@Data @EqualsAndHashCode(callSuper=false)
public final class Wedge
	extends Direction {

	/** Crescendo or diminuendo. */
	private final WedgeType type;
	/** End marker of this wedge. */
	private final WedgeEnd wedgeEnd = new WedgeEnd(this);
	/** Height of the opening in IS, or null for default. */
	private Float spread;

}

File: core/src/com/xenoage/zong/core/music/direction/Wedge.kt
package com.xenoage.zong.core.music.direction

import com.xenoage.zong.core.music.IS


/**
 * Class for crescendos and diminuendos.
 *
 * To create a wedge, create an instance of this class,
 * retrieve its end marker by calling [getWedgeEnd] and
 * place it anywhere after the instance of this class. The wedge ends
 * at the element at which the end marker is placed.
 */
class Wedge(
		/** Crescendo or diminuendo. */
		val type: WedgeType
) : Direction() {

	/** End marker of this wedge.  */
	val wedgeEnd = WedgeEnd(this)

	/** Height of the opening, or null for default. */
	var spread: IS? = null

}
